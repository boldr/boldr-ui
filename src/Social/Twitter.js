/* @flow */
import React from 'react';
import Icon from '../Icons';

type Props = {
  href: ?String,
  size: Number,
};

const Twitter = (props: Props) => {
  if (props.href) {
    return (
      <a href={props.href} target="_blank">
        <Icon kind="twitter" color="#1c3050" size={props.size} />
      </a>
    );
  } else {
    return <Icon kind="twitter" color="#1c3050" size={props.size} />;
  }
};

Twitter.defaultProps = {
  size: 24,
};
export default Twitter;
