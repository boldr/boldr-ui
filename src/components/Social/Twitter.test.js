import React from 'react';
import { shallow, mount } from 'enzyme';
import Twitter from './Twitter';

test('<Twitter />', () => {
  const wrapper = shallow(<Twitter href="a" />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
