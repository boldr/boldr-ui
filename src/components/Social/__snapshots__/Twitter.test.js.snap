// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Twitter /> 1`] = `
<a
  href="a"
  target="_blank"
>
  <Icon
    color="#1c3050"
    kind="twitter"
    size={24}
  />
</a>
`;
